components/ide/jetbrains/launcher/main.go

```diff
@@ -171,6 +172,33 @@ func main() {
 	serve(launchCtx)
 }
 
+func starWarmup() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var conn *grpc.ClientConn
+	var err error
+
+	for {
+		conn, err = dial(ctx)
+		if err == nil {
+			log.Debug("launcher: wait for tasks to finish before running warmup")
+			finished, err := waitForTasksToFinish(ctx, conn)
+			if err != nil {
+				log.WithError(err).Warn("launcher: failed to observe tasks completion")
+			}
+			if finished {
+				cancel()
```

Comment:
I am not sure why we need cancel at all, return here?

Reply:
now `checkTasks` return `nil` but it does not interrupt the outer loop in `waitForTasksToFinish` 🤔

Reply:
working now, it was missing a return